### DISCUSSION TITLE
Unknown Error in code when i add delay

### DISCUSSION THREAD

**Discussion by SandeepJagdev** [pschatzmann/arduino-audio-tools#492] [Q&A] [answered]
#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"
#include "AudioLibs/AudioA2DP.h"
#include "AudioLibs/Jupyter.h"
#include "AudioLibs/AudioEffectsSuite.h"

BluetoothA2DPSink a2dp_sink;
AudioKitStream kit;

AudioKitStream kit;
AudioEffects> effects(kit, 1, 1.0);
GeneratedSoundStream in(effects); 
// Write data to AudioKit in callback
void read_data_stream(const uint8_t *data, uint32_t length) {
    kit.write(data, length);
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  cfg.sd_active = false;
  kit.begin(cfg);

  // register callback
  a2dp_sink.set_stream_reader(read_data_stream, false);
  a2dp_sink.start("AudioKit");
  reset();
Delay effect(200, 0.5, 0.5, cfg.sample_rate);
effects.addEffect(effect);
}

void loop() {
  kit.processActions();
}

According to instructions i add delay in A2DP code.Then arduino show unknown error. Please tell me whats wrong.

**Comment by pschatzmann** [comment, marked as answer]
You pass a variable that was created on the stack to the effects. When the loop is executed this variable does not exist any more and bad things will happen!
Define it either as global variable or via new on the heap...
